### PR TITLE
Make Magnet Rise work in Inverse Battles

### PR DESCRIPTION
**Pull request by urkerab**
Also Telekinesis's ungroundedness with a Ring Target holder etc.

### CODE REVIEW

File: test/sim/misc/inversebattle.js [hunk 1]
```diff
@@ -79,6 +79,15 @@ describe('Inverse Battle', function () {
...
 
+	it('should not affect move-based immunities', function () {
+		battle.setPlayer('p1', {team: [{species: "Hariyama", ability: 'guts', moves: ['earthquake']}]});
+		battle.setPlayer('p2', {team: [{species: "Klefki", ability: 'prankster', moves: ['magnetrise']}]});
+		battle.makeChoices('move earthquake', 'move magnetrise');
+		battle.makeChoices('move earthquake', 'move magnetrise');
+		assert(battle.log[battle.lastMoveLine + 1].startsWith('|-immune|'));
+		assert.equal(battle.p2.active[0].hp, battle.p2.active[0].maxhp);
+	});
+
```
Comment (DaWoblefet) [inline test/sim/misc/inversebattle.js:90]:
```suggestion
	it('should not affect move-based immunities', function () {
		battle.setPlayer('p1', {team: [{species: "Hariyama", moves: ['earthquake']}]});
		battle.setPlayer('p2', {team: [{species: "Klefki", ability: 'prankster', moves: ['magnetrise']}]});
		battle.makeChoices();
		assert(battle.log[battle.lastMoveLine + 1].startsWith('|-immune|'));
		assert.fullHP(battle.p2.active[0]);
	});

```
Should simplify things a bit.

Comment (urkerab) [reply test/sim/misc/inversebattle.js:90]: Why don't you fix up all of the other tests while you're there?

Comment (DaWoblefet) [reply test/sim/misc/inversebattle.js:90]: Sure, I can do that.

**Comment by DaWoblefet** [comment]
Thanks urkerab!

**Pull request merged**